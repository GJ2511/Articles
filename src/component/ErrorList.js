import React from 'react';
import PropTypes from 'prop-types';

const parseErrorContent = (errors) => {
    return Object.keys(errors).map((item, i) => {
        return (
            <li key={i}>
                <span>
                    {item} {errors[item][0]}
                </span>
            </li>
        );
    });
};

function ErrorList({ errors }) {
    return (
        <div className="alert alert-danger" role="alert">
            <ul>{parseErrorContent(errors)}</ul>
        </div>
    );
}

ErrorList.propTypes = {
    errors: PropTypes.object,
};

export default ErrorList;
