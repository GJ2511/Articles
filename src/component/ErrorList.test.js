import React from 'react';

import ErrorList from './ErrorList';

const props = {
    errors: { ERROR: ['true'] },
};

describe('<ErrorList tests />', () => {
    test('match snapshot', () => {
        expect(shallow(<ErrorList {...props} />)).toMatchSnapshot();
    });
});
