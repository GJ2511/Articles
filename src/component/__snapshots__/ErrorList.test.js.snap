// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorList tests /> match snapshot 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  <ul>
    <li
      key="0"
    >
      <span>
        ERROR
         
        true
      </span>
    </li>
  </ul>
</div>
`;
